web/src/features/datasets/server/dataset-router.ts

```diff
@@ -620,6 +626,26 @@ export const datasetRouter = createTRPCRouter({
         `,
       );
 
+      const validatedTraceScores = traceScores.reduce((acc, ts) => {
```

Comment:
does it make sense to keep a function to transorm prisma score into zod-score at a central place? similar to how it is done for the models api